frappe/email/doctype/notification/notification.py

```diff
@@ -363,18 +392,28 @@ def get_receiver_list(self, doc, context):
 
 			# For sending messages to the owner's mobile phone number
 			if recipient.receiver_by_document_field == "owner":
-				receiver_list += get_user_info([dict(user_name=doc.get("owner"))], "mobile_no")
+				receiver_list += get_user_info([dict(user_name=doc.get("owner"))], user_field)
 			# For sending messages to the number specified in the receiver field
 			elif recipient.receiver_by_document_field:
-				receiver_list.append(doc.get(recipient.receiver_by_document_field))
+				data_field, child_field = _parse_receiver_by_document_field(
+					recipient.receiver_by_document_field
+				)
+				if child_field:
+					for d in doc.get(child_field):
+						if recv := field_extractor_func(d, data_field):
```

Comment:
Similar here, do confirm that a function is passed - otherwise fallback to the previous behaviour _maybe_?